src/HandleTextSelection.tsx

```diff
@@ -51,10 +52,29 @@ const HandleTextSelection: FunctionComponent<SelectionHandlerProps> = ({
         return domRectToSelectionRectangle(rectangle, region);
       })
       .filter(notNull);
-
-    onSelect({ text: selection.toString(), selectionRectangles });
+    const text = getSelectedText();
+    onSelect({ text, selectionRectangles });
   };
 
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if ((event.ctrlKey || event.metaKey) && event.key === 'c') {
+        event.preventDefault();
+        const selection = window.getSelection();
+        if (selection && selection.toString().trim() && ref.current) {
+          const text = getSelectedText();
+          navigator.clipboard.writeText(text);
+        }
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
```

Comment:
Would this not affect every event on the document? Even those that happen outside the text selection area (the pdf document)? Would this not affect a copy event that happens on the sidepanel for example?